src/base/api.lua

```diff
@@ -76,12 +76,15 @@
 ---
 -- Register a general-purpose includeExternal() call which works just like
 -- include(), but marks any containers created while evaluating the included
--- scripts as external.
+-- scripts as external. It also, loads the file regardless of how many times
+-- it has been loaded already.
 ---
 
 	function includeexternal(fname)
+		local fullPath = os.locate(fname, fname .. ".lua", path.join(fname, "premake5.lua"), path.join(fname, "premake4.lua"))
```

Comment:
Rather than copying this line, it would be better if we had a common function that could be called by `includeexternal()` and `include()` to get the full path. This could go in `_foundation.lua` and perhaps be called `premake.findProjectScript(name)` maybe?

That way if we need to change the search logic we can do it in one place. I could see this easily falling out of sync otherwise.